week3/homework/src/utilities.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const dataDealer = require('./data/data_dealer');
+const read = dataDealer.read;
+const write = dataDealer.write;
+const shortid = require('shortid');
+const TodoItem = require('./todo_obj_constructor');
+
+const validation = function(req) {
+  const reqBodyLength = Object.keys(req.body).length;
+  const reqToDo = req.body.todo || {};
+  let reqDescription = reqToDo.description || {};
+
+  if (typeof reqDescription === 'string') {
+    reqDescription = reqDescription.trim();
+  }
+
+  if (
+    reqBodyLength !== 1 ||
+        reqDescription === {} ||
+        typeof reqDescription !== 'string' ||
+        reqDescription.length === 0
+  ) {
+    return (reqDescription = '');
+  }
+  return reqDescription;
+};
+
+const findToDo = function(CurrentList, reqId) {
+  let toDo = CurrentList.find(todo => todo.id === reqId);
+  if (toDo === undefined) {
+    throw new Error(`The To-Do item with ID:${reqId} is not existed`);
+  }
+  return toDo;
+};
+
+const createToDo = function(req, res) {
+  const toDoDescription = validation(req);
+  if (toDoDescription === '') {
+    res.status(202).json({ Notification: 'The posted To-Do is not valid' });
+    return;
+  }
+  const todoObject = new TodoItem(shortid.generate(), toDoDescription, false);
```

Comment:
the `id` could be generated inside the `todo_obj_constructor.js` file since it's only related to id:)

Reply:
that makes sense: the constructor has to construct it with all what it needs. so also with ID.